storage/storage.go

```diff
@@ -390,7 +390,8 @@ func SeparateModTarget(ctx context.Context, body []byte, modID, name, modVersion
 			continue
 		}
 
-		err = copyModFileToArchZip(file, zipWriter, strings.TrimPrefix(file.Name, target+"/"))
+		encodedFileName := EncodeName(strings.TrimPrefix(file.Name, target+"/"))
+		err = copyModFileToArchZip(file, zipWriter, encodedFileName)
```

Comment:
As per Mircea's response on Discord: https://discord.com/channels/555424930502541343/830842478956642354/1198555511533092955

> Nope, that's for the files inside the zip

Reply:
Because `EncodeName` doesn't escape forward slashes, what I would do is:

```go
	return true, EncodeName(key), hex.EncodeToString(hash.Sum(nil)), int64(buf.Len())
```

Although I would appreciate an extra pair of eyes to double-check this.

Reply:
Only the filename needs to be encoded, not the whole key, so 

```go
	return true, fmt.Sprintf("/mods/%s/%s.smod", modID, EncodeName(filename)), hex.EncodeToString(hash.Sum(nil)), int64(buf.Len())`
```

and extracting the filename from the key creation, so it can be used there

```go
	filename := cleanName+"-"+target+"-"+modVersion
	key := fmt.Sprintf("/mods/%s/%s.smod", modID, filename)
```